dali/CMakeLists.txt

```diff
@@ -216,6 +229,26 @@ if (BUILD_PYTHON)
   add_subdirectory(pipeline/operators/python_function)
   # Get all python srcs
   add_subdirectory(python)
+
+  # prepare check-python target
+  set(CHECK_PYTHON_TARGET "check-python")
+  add_custom_target("${CHECK_PYTHON_TARGET}")
+
+  set(PYTHON_TARGET_PATH ${PROJECT_BINARY_DIR}/dali/python)
+  if($ENV{PYTHONPATH})
+    set(PYTHONPATH "${PYTHON_TARGET_PATH}:$ENV{PYTHONPATH}")
+  else()
+    set(PYTHONPATH "${PYTHON_TARGET_PATH}")
+  endif()
+
+  add_custom_command(
+    TARGET ${CHECK_PYTHON_TARGET}
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/qa/L0_python-self-test/
+    COMMAND PYTHONPATH=${PYTHONPATH} ${PROJECT_SOURCE_DIR}/qa/L0_python-self-test/test_no_setup.sh
+  )
+  add_dependencies(${CHECK_TARGET} ${CHECK_PYTHON_TARGET})
+  add_dependencies(${CHECK_PYTHON_TARGET} ${dali_python_lib})
+  add_dependencies(${CHECK_PYTHON_TARGET} ${dali_lib})
```

Comment:
I think `dali_python_lib` depends already on `dali_lib` so it is redundant.
Also user need to install python lib manually, and this step is not covered by this script.
So calling:
```
make -j
make check
```
Will not check latest python_lib.

Reply:
> Also user need to install python lib manually, and this step is not covered by this script.

This is handled by specifying PYTHONPATH pointing to the output directory with python module (pre wheel build). It will be searched first when script uses `from nvidia.dali import something`. That way you always use the latest output of the build without a need for install.